drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -121,98 +270,233 @@ local function assign_child_profile(device, child_ep)
+          if ep ~= main_endpoint then
+            if device:supports_server_cluster(clusters.OnOff.ID, ep) then
+              component_map[string.format("switch%d", current_component_number)] = ep
+            else
+              component_map[string.format("button%d", current_component_number)] = ep
+            end
+            current_component_number = current_component_number + 1
+          else
+            component_map["main"] = ep
+          end
+          component_map_used = true
+        else -- Create child devices for non-main endpoints
+          if ep ~= main_endpoint then -- don't create a child device that maps to the main endpoint
+            local name = string.format("%s %d", device.label, num_server_eps)
+            local child_profile = assign_child_profile(device, ep)
+            driver:try_create_device(
+                {
+                  type = "EDGE_CHILD",
+                  label = name,
+                  profile = child_profile,
+                  parent_device_id = device.id,
+                  parent_assigned_child_key = string.format("%d", ep),
+                  vendor_provided_label = name
+                }
+            )
+            current_component_number = current_component_number + 1
+          end
+        end
       end
     end
-  end
 
-  if num_server_eps > 1  then
-    -- If the device is a parent child device, then set the find_child function on init.
-    -- This is persisted because initialize switch is only run once, but find_child function should be set
-    -- on each driver init.
-    device:set_field(IS_PARENT_CHILD_DEVICE, true, {persist = true})
-  end
-
-  device:set_field(SWITCH_INITIALIZED, true)
-  -- The case where num_server_eps > 0 is a workaround for devices that have a
-  -- Light Switch device type but implement the On Off cluster as server (which is against the spec
-  -- for this device type). By default, we do not support Light Switch device types because by spec these
-  -- devices need bindings to work correctly (On/Off cluster is client in this case), so these device types
-  -- do not have a generic fingerprint and will join as a matter-thing. However, we have seen some devices
-  -- claim to be Light Switch device types and still implement their clusters as server, so this is a
-  -- workaround for those devices.
-  if num_server_eps > 0 and detect_matter_thing(device) == true then
-    local id = 0
-    for _, ep in ipairs(device.endpoints) do
-      -- main_endpoint only supports server cluster by definition of get_endpoints()
-      if main_endpoint == ep.endpoint_id then
-        for _, dt in ipairs(ep.device_types) do
-          -- no device type that is not in the switch subset should be considered.
-          if (ON_OFF_SWITCH_ID <= dt.device_type_id and dt.device_type_id <= ON_OFF_COLOR_DIMMER_SWITCH_ID) then
-            id = math.max(id, dt.device_type_id)
+    if num_server_eps > 1  then
```

Comment:
The assumptions that `num_server_eps > 1` means that the device is parent-child is no longer true since a n-button device would have 2 server eps but be represented with an MCD profile. You can just exchange this for a boolean that is set to true when `try_create_device` is called in the conditional above